app/models/validators/parent_task_validator.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+##
+# Validates the parent task
+#
+# Usage example: For the parent field, calls the built-in PresenceValidator and
+# ParentTaskValidator with argument DistributionTask when the record is being created:
+#   validates :parent, presence: true, parentTask: { task_type: DistributionTask }, on: :create
+
+class ParentTaskValidator < ActiveModel::Validator
```

Comment:
Tempted to suggest
```suggestion
class ParentTaskTypeValidator < ActiveModel::Validator
```
but it feels a bit nitpicky - up to you!